pkg/cloudprovider/amifamily/ami.go

```diff
@@ -193,23 +193,41 @@ func (p *AMIProvider) fetchAMIsFromEC2(ctx context.Context, amiSelector map[stri
 	return output.Images, nil
 }
 
-func getFilters(amiSelector map[string]string) []*ec2.Filter {
+func getFiltersAndOwners(amiSelector map[string]string) ([]*ec2.Filter, []*string) {
 	filters := []*ec2.Filter{}
+	owners := []*string{}
 	for key, value := range amiSelector {
-		if key == "aws-ids" {
+		switch key {
+		case "aws-ids":
 			filterValues := functional.SplitCommaSeparatedString(value)
 			filters = append(filters, &ec2.Filter{
 				Name:   aws.String("image-id"),
 				Values: aws.StringSlice(filterValues),
 			})
-		} else {
+		case "aws:owners":
+			ownerValues := functional.SplitCommaSeparatedString(value)
+			for _, owner := range ownerValues {
+				owners = append(owners, &owner)
+			}
+		case "aws:name":
+			filters = append(filters, &ec2.Filter{
+				Name:   aws.String("name"),
+				Values: []*string{aws.String(value)},
+			})
+		default:
 			filters = append(filters, &ec2.Filter{
 				Name:   aws.String(fmt.Sprintf("tag:%s", key)),
 				Values: []*string{aws.String(value)},
 			})
 		}
 	}
-	return filters
+	if owners == nil {
+		self := aws.String("self")
+		amazon := aws.String("amazon")
+		owners = []*string{self, amazon}
+	}
```

Comment:
nit: you can collapse this into one line if you'd like.
```suggestion
	if owners == nil {
		owners = []*string{aws.String("self"), aws.String("amazon")}
	}
```